guides/getting-started/connecting-to-kuri.md

```diff
@@ -31,19 +31,72 @@ Through bluetooth the phone looks for Kuri and, if more than one is found, will
+and Kuri should emit a pleasant chime around that time), then try again
+(up to 10 times or so--unfortunately, this part got a bit flaky during the hasty
+conversion to mini-cloud/no-cloud).
+
+If wifi still will not connect, try re-starting setup and trying again.
+
+If that still fails, try doing a
+[Manual factory reset](https://helpcenter.heykuri.com/hc/en-us/articles/360001548488-Manual-factory-reset),
+and then try setup again.
+
+## Connecting to Kuri again
+
+After this is completed, the app connects to Kuri so that the rest of setup can be finished.  These steps include creating a map of the house, and learning to control him through the app.  
 
 ![](/assets/images/getting-started/Screenshot_Kuri_20181008-133104.png)
 
+If your app gets stuck here, it may be because your Kuri needs to update
+to the newest Kuri software version.
+
+## Kuri software update
+If your Kuri is not on the newest software version (which is the case for most
+new / newly-refurbished Kuris), the app should trigger Kuri to
+query the update server and update his software at this point (which can take from
+several minutes to about half an hour, depending on the speed of your connection),
+but sometimes this attempt fails, especially if the wifi network that Kuri
+is on has an annoying firewall / network configuration that prevents
+computer-to-computer connections, or blocks MQTT messages from getting through.  
+
+To get your Kuri to update separately from using the app, take him to
+a network without restrictions, leave him on and charging on the dock,
+and he should attempt to auto-update every hour, 15 minutes after the hour.
+
+If he starts successfully updating, the power light should go from solid to
+blinking.  It will stay blinking while the update is downloading / installing.
+When that is finished (somewhere between about 5 and 30 minutes later),
+Kuri will automatically reboot.  The power light should eventually turn solid
+again (and Kuri should make his startup chime noise).
+
+Note that only the newest software release allows you to ssh into your Kuri;
+ssh is turned off in older software releases.
+
+Once Kuri has updated successfully to the newest software release, he will
+stop checking for new updates (since there won't be any further releases
+after the current one).  
+
 ## Finding Kuri's hostname
 Kuri's hostname is based upon the serial number of the robot which is printed on the box as well as on the bottom of the robot.  The hostname is built with a base `kuri-0000` with the addition of the last numbers and letters from the serial number.
 
 For example, if Kuri's serial number is `KR118160000abc`, the hostname is `kuri-0000abc`.
 
+The exception is some (but not all) robots that start with 'KR01744', which had a different
+serial number scheme (don't ask).  If you have one such, take the last numbers from the
+serial number and convert that number to hexadecimal.  So, for serial number `KR017440000149`,
```

Comment:
If they were reprovisioned properly, I think the hostname should just be the serial. @po1, could you verify?

Reply:
Definitely not true for any of the robots I've QA'd. Easiest way to figure out the hostname is to grab a phone, go to Settings > Bluetooth and look for anything that starts with `kuri-`. That name is the hostname of the robot.

Reply:
The HFUs will have a `kuri-000xx` hex host name with `xx` varying loosely from 30 to bf

Later units and all reprovisioned robots will have `kuri-0000xxx` as a hostname, with the same digits as the last 7 digits of their S/N

Reply:
Most HFUs that are out there were not reprovisioned